### PR TITLE
Cpp pipeline

### PR DESCRIPTION
**Pull request by gearama**
ability to switch to vcpkg release versioon # ,

### CODE REVIEW

**Comment by gearama** [comment]
@mikeharder

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:107]: I'm not sure this is the correct approach.  It appears you are syncing the repo to an older tag, which I don't think will work for several reasons:

1. We want to use the perf tests from main, but use older versions of the SDK packages.
2. We want the ability to mix-and-match package versions.  Say Storage in main and Core from an older package, or vice versa, or both Storage and Core from older packages.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Program.cs
Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Program.cs:148]: for some reason VS decides those parans are superfluous , but not for the thing above

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/tests.yml
Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/tests.yml:75]: 1.7.2 is the currently released version

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Program.cs
Comment (mikeharder) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Program.cs:148]: Can you please revert this change?  Even if VS doesn't like these parens, I don't think they should be removed in an unrelated PR.

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:107]: That wwouldn't work for cpp , the code is static built thus we cannot load dlls at runtime how we like.

Comment (mikeharder) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:107]: I understand (at a high level) how vcpkg works, and I think we should be able to make the C++ perf apps support arbitrary package versions, the same way a customer app can choose arbitrary versions of our packages, right?  This may require re-architecting how the perf apps are written and compiled in our repo.

Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:107]: i was thinking about it i i came to the same realization. i'll revert the branch checkout part

Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:107]: all we have to do is generate this https://github.com/microsoft/vcpkg/issues/21357

**Comment by gearama** [comment]
@mikeharder

**Comment by gearama** [comment]
@mikeharder @benbp

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:195]:
```suggestion
```

Looks like this code is now unused and can be removed

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:81]: The naming could be improved, but `IterationResults.PackageVersions` should be populated with the **runtime** package versions.  The `packageVersions` passed into `RunAsync()` is the **requested** package versions.

So the next question for C++ is how to **verify** which package versions are being used at runtime.  In .NET this is implemented in the perf framework by inspecting the assemblies loaded at runtime.  In Java, JS, and Python this is implemented using package manager commands (e.g. `mvn depdendency:list`).

Reporting the runtime package versions is almost as important as modifying the versions themselves, since it's so easy to accidentally test the wrong versions.

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (mikeharder) [inline tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:95]:
```suggestion
        private async Task<bool> UpdatePackageVersions(IDictionary<string, string> packageVersions)
```

**Review by mikeharder — CHANGES_REQUESTED** [review]

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:195]: done

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (gearama) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:95]: done

File: tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs
Comment (mikeharder) [reply tools/perf-automation/Azure.Sdk.Tools.PerfAutomation/Cpp.cs:195]: I am still seeing this code in the PR.  Did you intend to push a change that removes this?

**Comment by gearama** [comment]
@mikeharder pushed new changes, sorry for the delay , i had a release yesterday and themn my machine got borked by a windows update so i had to reinstall a bunch of stuff just to get things moving

**Comment by gearama** [comment]
@mikeharder can you  please take a look ? i reached a point where i am blocked by this PR getting in. It doesn't need to be perfect as i suspect i'll need to make other changes over time to support the other services such as  storage and keyvault

**Comment by mikeharder** [comment]
@gearama: I can review this again today.  Before this is merged, I want to make sure it's actually working as intended, by adding a test that uses Azure.Core, and then intentionally introduce a perf regression in main, and make sure the test results capture this.

So I've got a draft PR here adding this test, and I'll combine this with your tools PR to validate this end-to-end.

https://github.com/Azure/azure-sdk-for-cpp/pull/4012

**Review by mikeharder — APPROVED** [review]

**Pull request merged**
